### PR TITLE
Enable Stylus support

### PR DESCRIPTION
**Pull request by bitpshr**
Adds a `--stylus` option so typings can be generated for CSS modules written in Stylus

### CODE REVIEW

File: src/generateTypings.ts [hunk 1]
```diff
@@ -4,36 +4,51 @@ import * as path from 'path';
...
 			}
 
-			resolve(buffer);
+			resolve(content.toString());
 		});
 	});
 }
 
-function write(name: string, content: { formatted: string }) {
+function compileStylus(content: string): Promise<string> {
```
Comment (tomdye) [inline src/generateTypings.ts]: Would be good to walk through this with you to see how the process fits together, would this not be better in a `compileStylus` file? Doesn't seem like a direct concern of `generateTypings`. Though that's just my opinion.

File: src/generateTypings.ts [hunk 2]
```diff
@@ -4,36 +4,51 @@ import * as path from 'path';
...
+	return new Promise((resolve, reject) => {
+		stylus.render(content, (err: Error, css: string) => {
+			if (err) {
+				reject(err);
+				return;
```
Comment (tomdye) [inline src/generateTypings.ts:75]: no need to return here

File: tests/unit/main.ts [hunk 3]
```diff
@@ -58,4 +67,18 @@ describe('main', () => {
 			assert.strictEqual(writeFileStub.getCall(0).args[0], 'test.css.d.ts');
 		});
 	});
+
+	it('should create a Stylus CSS module declaration file', () => {
+		const writeFileStub: sinon.SinonStub = sandbox.stub(fs, 'writeFile', function (name: string, content: string, done: Function) {
```
Comment (tomdye) [inline tests/unit/main.ts]: might look a bit tidier if you imported `{ SinonStub }` from sinon as well as the default.

**Review by tomdye — CHANGES_REQUESTED** [review]

**Comment by agubler** [comment]
@Tomdye have your review feedback been addressed?

**Comment by dylans** [comment]
Given switch to postcss, closing this out.

**Pull request closed without merging**